conda_build/index.py

```diff
@@ -29,7 +28,6 @@
 local_output_folder = ""
 cached_channels = []
 _channel_data = {}
-deprecated.constant("24.1", "24.7", "channel_data", _channel_data)
```

Comment:
`channel_data` was previously postponed since the global variable (`_channel_data`) was still in use. What would it take for us to remove this global variable and instead only use a local variable?